apps/src/blockly/googleBlocklyWrapper.js

```diff
@@ -319,9 +319,7 @@ function initializeBlocklyWrapper(blocklyInstance) {
   };
 
   blocklyWrapper.BlockSvg.prototype.isUnused = function() {
-    const isTopBlock = this.previousConnection === null;
-    const hasParentBlock = !!this.parentBlock_;
-    return !(isTopBlock || hasParentBlock);
+    return this.disabled;
```

Comment:
I tested this in dance party and it works as expected, unconnected blocks are not included in the generated code. Do we know why we used the previous logic?

Reply:
I think this might be okay? Here's some interesting history I found:
* It looks like we originally created this to prevent generating code on blocks that were being dragged. At that point we actually just looked at `this.disabled` for other cases: https://github.com/code-dot-org/code-dot-org/commit/a34bbf56d4102b3f594aff9c233877fa3cea40b1#diff-14806d007239a925c2a884931f6fbe47f036be5597e29a73cc324340e29887e9
* We moved from `this.disabled` to checking top/parent blocks here, but the PR doesn't make it totally clear why: https://github.com/code-dot-org/code-dot-org/pull/43908/files#diff-14806d007239a925c2a884931f6fbe47f036be5597e29a73cc324340e29887e9
* Later, the dragging part of the logic was removed because it was actually causing a bug: https://github.com/code-dot-org/code-dot-org/pull/44027/files

We might want to check that blocks are not being disabled in toolbox editing mode in the levelbuilder interface.

Reply:
it looks like everything works the same in levelbuilder mode!

Reply:
Thank you for double-checking!